RDS 64 - UI Basic Gameplay

chentra99 opened the pull request:
https://jira.internal.teamriders.dev/browse/RDS-64

- Hooked up RaceManager to toggle HUD visibility when race starts
- Connected UI elements to simulated values from RacerComponent
- Position, lap, boost, speed, and timer now update in real time

chentra99 committed:
feat: (ui) show InGameHUD on race load with dummy player stats

- Hooked up RaceManager to toggle HUD visibility when race starts
- Connected UI elements to simulated values from RacerComponent
- Position, lap, boost, speed, and timer now update in real time

chentra99 committed:
feat: UI Basic Gameplay

- Displays HUD when race starts

jtbui20 commented on Assets/Scripts/RidersGame/RaceManager/RaceHostController.cs:
```diff
@@ -111,6 +111,13 @@ public async void OnRaceMeetSetupComplete(List<RiderSelection> selectedRiders)
 
             sessionRaceMeetController.AssignFields(sessionMeet, allRiders, OnRaceMeetComplete);
 
+            // 👇 Show HUD just before race starts
```
Move to RaceEventController into StartRace

jtbui20 commented on Assets/Scripts/RidersGame/RaceManager/RaceManager.cs:
Probably rename to something UI related and move to the UI folder as it is mostly a UI related component

chentra99 committed:
Merge branch 'dev' into feature/rds-64-ui-gameplay-basic

chentra99 committed:
feat: connect GetPlayerInfo to live gameplay data

- GetPlayerInfo now fetch real-time data from RacerComponent and BoostController
- Integrated RacerComponent for real-time position, lap, and speed
- Integrated BoostController for current boost value and usage state

jtbui20 commented:
Broken, work done over-reaches scope, taking over
